lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -296,9 +305,37 @@ void StagesSubject::stageChanged(
             const TfToken nameToken = changedPath.GetNameToken();
             if (nameToken == UsdGeomTokens->xformOpOrder || UsdGeomXformOp::IsXformOp(nameToken)) {
                 Ufe::Transform3d::notify(ufePath);
+#ifdef UFE_V2_FEATURES_AVAILABLE
+                anyUfeV2NotifSent = true;
+#endif
+            }
+        }
+    }
+
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    // If we get here and didn't send any UFE notif, but yet have paths
+    // then we'll send a catch-all subtree invalidate notification.
+    if (!anyUfeV2NotifSent) {
+        SdfPathVector allPaths = (SdfPathVector)notice.GetResyncedPaths();
```

Comment:
We're already looping over resync'ed paths at line 199.  Why can't the catch-all SubtreeInvalidate be in that loop?

Reply:
Because at lines 319-323 I merge all the paths together into a single SdfPathVector and then a tline 324 it removes any that are descendant paths. This way if you have two paths: /tree/one and /tree, we just end up with /tree.